### PR TITLE
Handle null values in istio-discovery template

### PR DESCRIPTION
**Pull request by Stevenjin8**
Fix for #45855

I ran `helm install --dry-run -f config.yaml ...` and  `helm lint -f config.yaml ...` in every chart. I fixed the null traversal in `istio-discovery` and `make gen` fixed the one in `istiod-remote`. Is there any other way I can look for similar issues?

```yaml
# config.yaml
global:
  logAsJson: true
  proxy:
    tracer: "datadog"
```

### CODE REVIEW

**Comment by howardjohn** [comment]
LGTM!

I don't know how to find a full set but my IDE (intellij) hinted me to `lightstep` and `stackdriver` sections in the same area as having the same issue

**Review by howardjohn — APPROVED** [review]

**Comment by Stevenjin8** [comment]
@howardjohn Same issue occurs with `lightstep` and `stackdriver`, but unlike `datadog`, there aren't always default values. 

`lightstep` doesn't have default values for `.address` and `.accessToken`, which seems pretty reasonable to me. `stackdriver` is only missing the `.debug` default. I'm leaning on setting it to false by default but would like input.

**Comment by howardjohn** [comment]
false by default is good

File: manifests/charts/istio-control/istio-discovery/templates/configmap.yaml [hunk 1]
```diff
@@ -47,25 +47,17 @@
...
+          maxNumberOfAttributes: {{ (($.Values.global.tracer).stackdriver).maxNumberOfAttributes | default "200" }}
           # The global default max number of annotation events per span.
-          maxNumberOfAnnotations: {{ $.Values.global.tracer.stackdriver.maxNumberOfAnnotations | default "200" }}
-        {{- end }}
-        {{- if $.Values.global.tracer.stackdriver.maxNumberOfMessageEvents }}
+          maxNumberOfAnnotations: {{ (($.Values.global.tracer).stackdriver).maxNumberOfAnnotations | default "200" }}
           # The global default max number of message events per span.
-          maxNumberOfMessageEvents: {{ $.Values.global.tracer.stackdriver.maxNumberOfMessageEvents | default "200" }}
-        {{- end }}
+          maxNumberOfMessageEvents: {{ (($.Values.global.tracer).stackdriver).maxNumberOfMessageEvents | default "200" }}
```
Comment (Stevenjin8) [inline manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:60]: There is a weird edge case here where:
```
"0" | default "200" -> 0
0 | default "200" -> 200
```

Comment (Stevenjin8) [reply manifests/charts/istio-control/istio-discovery/templates/configmap.yaml:60]: Also wondering if this counts as a breaking change

**Comment by Stevenjin8** [comment]
@howardjohn This got automerged after my last commit, could you take a look when you get a sec? I'll label with `do not merge` next time.

**Comment by howardjohn** [comment]
Yep this looks good :+1: thanks

**Pull request merged**